File: CS/AggregationSample/Form1.cs
using DevExpress.XtraCharts;
using System;
using System.Collections.Generic;
using System.Windows.Forms;

namespace AggregationSample {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }

        #region #ConfigureChart
        private void Form1_Load(object sender, EventArgs e) {
            chart.Series.Add(GenerateSeries(10000));

            XYDiagram diagram = chart.Diagram as XYDiagram;
            if (diagram == null) return;

            diagram.AxisX.NumericScaleOptions.AggregateFunction = AggregateFunction.Average;
            diagram.AxisX.NumericScaleOptions.ScaleMode = ScaleMode.Automatic;
            diagram.AxisX.NumericScaleOptions.AutomaticMeasureUnitsCalculator = new CustomNumericMeasureUnitCalculator();

            diagram.AxisY.WholeRange.AlwaysShowZeroLevel = false;
        }
        #endregion #ConfigureChart

        Series GenerateSeries(int pointCount) {
            Series series = new Series {
                Name = "Random data",
                View = new SideBySideBarSeriesView()
            };
            Random generator = new Random();
            for (int i = 0; i < pointCount; ++i) {
                series.Points.Add(new SeriesPoint((double)i, generator.NextDouble()));
            }
            return series;
        }
    }

    #region #InterfaceImpl
    class CustomNumericMeasureUnitCalculator : INumericMeasureUnitsCalculator {
        public double CalculateMeasureUnit(
                IEnumerable<Series> series, 
                double axisLength, 
                int pixelsPerUnit, 
                double visualMin, 
                double visualMax, 
                double wholeMin, 
                double wholeMax) {
            double visualRange = visualMax - visualMin;
            return Math.Ceiling(visualRange / 20);
        }
    }
    #endregion #InterfaceImpl
}

File: VB/AggregationSample/Form1.vb
Imports DevExpress.XtraCharts
Imports System
Imports System.Collections.Generic
Imports System.Windows.Forms

Namespace AggregationSample
    Partial Public Class Form1
        Inherits Form

        Public Sub New()
            InitializeComponent()
        End Sub

        #Region "#ConfigureChart"
        Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
            chart.Series.Add(GenerateSeries(10000))

            Dim diagram As XYDiagram = TryCast(chart.Diagram, XYDiagram)
            If diagram Is Nothing Then
                Return
            End If

            diagram.AxisX.NumericScaleOptions.AggregateFunction = AggregateFunction.Average
            diagram.AxisX.NumericScaleOptions.ScaleMode = ScaleMode.Automatic
            diagram.AxisX.NumericScaleOptions.AutomaticMeasureUnitsCalculator = New CustomNumericMeasureUnitCalculator()

            diagram.AxisY.WholeRange.AlwaysShowZeroLevel = False
        End Sub
        #End Region ' #ConfigureChart

        Private Function GenerateSeries(ByVal pointCount As Integer) As Series
            Dim series As Series = New Series With {.Name = "Random data", .View = New SideBySideBarSeriesView()}
            Dim generator As New Random()
            For i As Integer = 0 To pointCount - 1
                series.Points.Add(New SeriesPoint(CDbl(i), generator.NextDouble()))
            Next i
            Return series
        End Function
    End Class

    #Region "#InterfaceImpl"
    Friend Class CustomNumericMeasureUnitCalculator
        Implements INumericMeasureUnitsCalculator

        Public Function CalculateMeasureUnit(
                ByVal series As IEnumerable(Of Series),
                ByVal axisLength As Double,
                ByVal pixelsPerUnit As Integer,
                ByVal visualMin As Double,
                ByVal visualMax As Double,
                ByVal wholeMin As Double,
                ByVal wholeMax As Double) As Double Implements INumericMeasureUnitsCalculator.CalculateMeasureUnit
            Dim visualRange As Double = visualMax - visualMin
            Return Math.Ceiling(visualRange / 20)
        End Function
    End Class
    #End Region ' #InterfaceImpl
End Namespace
